src/VisualStudio/IntegrationTest/TestUtilities/TemporaryTextFile.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+
+namespace Microsoft.VisualStudio.IntegrationTest.Utilities
+{
+    /// <summary>
+    /// Provides methods for supporting temporary files within Roslyn tests.
+    /// </summary>
+    public class TemporaryTextFile : IDisposable
+    {
+        private string _fileName;
+        private string _content;
+        private string _path;
+
+        public TemporaryTextFile(string fileName, string content)
+        {
+            _fileName = fileName;
+            _content = content;
+            _path = Path.Combine(Path.GetTempPath(), "RoslynTests", "TemporaryTextFile");
```

Comment:
Do we have a better place to put these? Some sort of test directory? @tannergooding might have already had a solution for this...?